pkg/controller/nmstate/nmstate_controller_test.go

```diff
@@ -0,0 +1,64 @@
+package nmstate
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var _ = Describe("NMstate controller reconcile", func() {
+	var (
+		cl                  client.Client
+		reconciler          ReconcileNMstate
+		existingNMstateName = "nmstate01"
+		nmState             = nmstatev1alpha1.NMstate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: existingNMstateName,
+			},
+		}
+	)
+	BeforeEach(func() {
+		s := scheme.Scheme
+		s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion,
+			&nmstatev1alpha1.NMstate{},
+		)
+
+		objs := []runtime.Object{&nmState}
+
+		// Create a fake client to mock API calls.
+		cl = fake.NewFakeClientWithScheme(s, objs...)
+
+		reconciler.client = cl
+	})
+	Context("when nmstate is not found", func() {
+		var (
+			request reconcile.Request
+		)
+		BeforeEach(func() {
+			request.Name = "not-present-nmstate"
+		})
+		It("should return empty result", func() {
+			result, err := reconciler.Reconcile(request)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result).To(Equal(reconcile.Result{}))
+		})
+	})
+	Context("when nmstate is found", func() {
+		var (
+			request reconcile.Request
+		)
+		It("should return a Result with RequeueAfter set (trigger re-reconciliation)", func() {
+			result, err := reconciler.Reconcile(request)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result).To(Equal(reconcile.Result{RequeueAfter: nmstateRefresh}))
```

Comment:
ditto, here we will check that we deploy all the manifests, checking using FakeClient.Get to get them, but maybe we can just merge this like that.